release_automation.sh

```diff
@@ -343,9 +343,10 @@ ohai "Update GitHub organization and gutenberg-mobile ref"
 test -f "Podfile" || abort "Error: Could not find Podfile"
 sed -i'.orig' -E "s/wordpress-mobile(\/gutenberg-mobile)/$MOBILE_REPO\1/" Podfile || abort "Error: Failed updating GitHub organization in Podfile"
 sed -i'.orig' -E "s/gutenberg :(commit|tag) => '(.*)'/gutenberg :commit => '$GB_MOBILE_PR_REF'/" Podfile || abort "Error: Failed updating gutenberg ref in Podfile"
+execute "bundle" "install"
```

Comment:
My understanding is that `bundle install` is executed within `rake dependencies`, and thus leads to the failure when the pod repo is outdated. I believe we should remove this line.

Reply:
Now that I reread this, I believe I am mistaken. I am likely confusing `bundle install` and `pod install`. I often conflate the two tools.

Reply:
Good catch, I've just verified the [`Rakefile`](https://github.com/wordpress-mobile/WordPress-iOS/blob/d7dad2e5342439d0e83111d6e10c7e33afeab373/Rakefile#L26) and I confirm that it does this step. Although, we're going to run `bundle exec pod repo update` before running `rake dependencies` so the dependencies won't be up-to-date, probably this won't have any side effect, wdyt?

Reply:
I've just removed the `bundle install` step in [this commit](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/pull/52/commits/dafe490e30fad3e3aa37b994f2d5494034570203).

Reply:
Yeah, I'm not sure if `pod repo update` has a dependency on `bundle install` or not. I suppose we can try without it and see what occurs.

Reply:
Agreed, let's go without it, most likely it won't introduce errors 🤞 .

Reply:
I'm going to revert the change because I've recently bumped into the same issue when cutting a betafix:
```
==> Create release branch in WordPress-iOS
Switched to a new branch 'gutenberg/integrate_release_1.58.1'
==> Update GitHub organization and gutenberg-mobile ref
Warning: the running version of Bundler (2.2.10) is older than the version that created the lockfile (2.2.21). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.2.21`.
Could not find proper version of rake (12.3.3) in any of the sources
Run `bundle install` to install missing gems.

Failed during: rake dependencies
```

In this case, we need to run `bundle install` before executing `rake`.

Reply:
Heads up that I created [this PR](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/pull/58) to address the problem with Gem dependencies.